### PR TITLE
Uploaded tictactoe and tictactoe with ai

### PR DESCRIPTION
**Pull request by MoSBanapple**

### CODE REVIEW

File: tictactoe.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        (x,y) = tuple(read)
+        x = int(x)
+        y = int(y)
+        while (not board.Mark(turn, x, y)):
+            read = input("Invalid move. Please input again.\n")
+            (x,y) = tuple(read)
+            x = int(x)
+            y = int(y)
+        print()
+        counter = counter + 1
```
Comment (qiwzhang) [inline tictactoe.py]: counter += 1

File: tictactoe.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        
+if __name__ == '__main__':
+    board = Board()
+    turn = 2;
+    counter = 0;
+    while (board.Check() == 0 and counter < 9):
+        if (turn == 2):
+            turn = 1
+        else:
+            turn = 2
```
Comment (qiwzhang) [inline tictactoe.py]: turn = counter % 2  + 1

File: tictactoe.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        board.PrintGrid()
+        read = input('Player ' + str(turn) + '\'s turn.\n')
+        (x,y) = tuple(read)
+        x = int(x)
+        y = int(y)
```
Comment (qiwzhang) [inline tictactoe.py]: use a function for line 67 - 69 so it can be shared by line 72-74

File: tictactoe.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                    row += 'X'
+            print(row)
+            row = ''
+        
+        
+if __name__ == '__main__':
+    board = Board()
+    turn = 2;
+    counter = 0;
+    while (board.Check() == 0 and counter < 9):
```
Comment (qiwzhang) [inline tictactoe.py]: why < 9?

Comment (MoSBanapple) [inline tictactoe.py]: 9 is the maximum number of turns.

File: tictactoe.py [hunk 5]
```diff
@@ -51,31 +53,27 @@ def PrintGrid(self):
                     row += 'X'
             print(row)
             row = ''
-        
+
+def SplitInput(read):
```
Comment (qiwzhang) [inline tictactoe.py]: ReadInput
add function comment please

File: tictactoe.py [hunk 6]
```diff
@@ -51,31 +51,27 @@ def PrintGrid(self):
...
-            turn = 2
         board.PrintGrid()
-        read = input('Player ' + str(turn) + '\'s turn.\n')
-        (x,y) = tuple(read)
-        x = int(x)
-        y = int(y)
-        while (not board.Mark(turn, x, y)):
+        read = input('Player ' + str(counter%2 + 1) + '\'s turn.\n')
+        (x,y) = SplitInput(read))
+        while (not board.Mark(counter%2 + 1, x, y)):
```
Comment (qiwzhang) [inline tictactoe.py]: it is better to keep "turn" variable.  it is easy to read

File: tictactoe.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+"""
+Contains the tic tac toe board and functions that affect it
+"""
+class Board:
+
+    def __init__(self):
+        self.grid = []
+        for iter in range(3):
+            self.grid.append([])
```
Comment (qiwzhang) [inline tictactoe.py]: repleace line 12, 13, 14 with self.grid.append([0, 0, 0])

File: tictactoe.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            return False
+        self.grid[x - 1][y - 1] = player
+        return True
+        
+    """
+    Prints the board.
+    """
+    def PrintGrid(self):
+        row = ''
+        for array in self.grid:
```
Comment (qiwzhang) [inline tictactoe.py]: define a dict as

char = {0: '-', 1: '0', 2: 'X'}

then use it as
for item in array:
   raw += char[item]

File: tictactoe.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            print(row)
+            row = ''
+"""
+Reads an input of two numbers and splits it into a tuple of two integers.
+"""
+def ReadInput(read):
+    (x,y) = tuple(read)
+    x = int(x)
+    y = int(y)
+    return (x,y)
```
Comment (qiwzhang) [inline tictactoe.py]: replace line 61, 63 with
return (int(x), int(y))

**Pull request merged**